Update Eventing Overview Documentation (#14834)
kyma-project/kyma: docs/01-overview/main-areas/eventing/evnt-01-overview.md

title: Overview
---

Eventing, in general, consists of event producers (or publishers) and consumers (or subscribers) that send events to or receive events from an event processing backend. The PubSub messaging pattern, which is the eventing paradigm used in Kyma, does not send messages directly to the subscribers but instead sends them to particular topics (or event types) without any knowledge of existing subscribers. The diagram shows a high level overview of PubSub communication:

![PubSub](./assets/pubsub.svg)

The key benefits of PubSub are:
- communication between loosely coupled components.
- scale publishers and subscribers easily.
- asynchronous communication.

## Eventing in Kyma

> Eventing in Kyma is used to send and receive events from applications. For example, you can subscribe to events from an external application, and when the user performs an action there, you can trigger your Function or microservice. To subscribe to events, you need to use the Kyma [Subscription CR](../../../05-technical-reference/00-custom-resources/evnt-01-subscription.md). 

Reviewer: Maybe we should focus on the fact that Eventing in kyma is not meant to introduce a new backend. It provides a way to simplify publishing and subscribe to events. The backend for this is based on proven technology (in oss nats jetstream)

[NATS JetStream](https://docs.nats.io/) is the default backend used in Kyma to process and send events to subscribers. Read [Eventing architecture](../../../05-technical-reference/00-architecture/evnt-01-architecture.md) to know more about the eventing flow in Kyma.

## Glossary for Eventing

- **Event Types**

>   - `Legacy Events`: These are events or messages published to Kyma, that do not conform to the [Cloud Events specification](https://cloudevents.io/). All legacy events published to Kyma are always converted to Cloud Events.

Reviewer: ```suggestion
  - `Legacy Events`: These are events or messages published to Kyma, that do not conform to the [Cloud Events specification](https://cloudevents.io/). All legacy events published to Kyma are always converted to Cloud Events. The legacy event format is deprecated.
```

> title: Overview

Author: Yes will do 👍

---

> Eventing, in general, consists of event producers (or publishers) and consumers (or subscribers) that send events to or receive events from an event processing backend. The PubSub messaging pattern, which is the eventing paradigm used in Kyma, does not send messages directly to the subscribers but instead sends them to particular topics (or event types) without any knowledge of existing subscribers. The diagram shows a high level overview of PubSub communication:

Author: I would need to rewrite that sentence since its slightly misleading.

![PubSub](./assets/pubsub.svg)

The key benefits of PubSub are:
- communication between loosely coupled components.

> - scale publishers and subscribers easily.

Reviewer 2: ```suggestion
- Easy scaling of publishers and subscribers
```

keep conistent phrasing (in this case, nominal style) in all list items.

> - asynchronous communication.

Reviewer 2: ```suggestion
- Asynchronous communication
```
Is this another kind of communication than the one between loosely coupled components? 
Or could those be combined as "Asynchronous communication between loosely coupled components"?

Author: Yes its the same. I will combine them

## Eventing in Kyma in a nutshell

> The objective of Eventing in Kyma is to simplify the process of publishing and subscribing to events. Kyma uses proven Eventing backend technology such as [NATS JetStream](https://docs.nats.io/), so that the user can achieve a seamless experience with their end-to-end business flow. The user does not have to implement or integrate any intermediate backend or protocol. Eventing in Kyma from a user's perspective works as follows:

Reviewer 2: Capitalize Eventing if it's the name of the Kyma component, use lowercase if it's "eventing" in general.

```suggestion
The objective of Eventing in Kyma is to simplify the process of publishing and subscribing to events. Kyma uses proven eventing backend technology such as [NATS JetStream](https://docs.nats.io/), so that the user gets a seamless experience with their end-to-end business flow. The user does not have to implement or integrate any intermediate backend or protocol. Eventing in Kyma from a user's perspective works as follows:
```

>  - Offer an HTTP end point, for example a lambda function to receive the events.

Reviewer 2: is "lambda function" something else than the usual Kyma Functions?
If no, write "[Function](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#capitalized-terms)".

Author: Its just an example. I will use Function 👍

 - Specify the events the user is interested in using the Kyma [Subscription CR](../../../05-technical-reference/00-custom-resources/evnt-01-subscription.md).

>  - Send Legacy or [Cloud Events](https://cloudevents.io/) to configured HTTP end points on our [Event Publishing Proxy](https://github.com/kyma-project/kyma/tree/main/components/event-publisher-proxy) service.

Author: It does not really give more description about legacy events to the user. Do I still add it?

>    - `/publish` for Cloud Events.

Author: These are endpoints at which the user can publish events to depending on weather he send cloud events or legacy events. And yes, these are subpoints to the previous point where we talk about sending legacy/cloud events

   - `<application_name>/v1/events` for Legacy events.
 
Read [Eventing architecture](../../../05-technical-reference/00-architecture/evnt-01-architecture.md) to know more about the eventing flow in Kyma.

> ## Glossary for Eventing

Author: That sounds like a good idea. But I think it doesn't fit in with those items since they are high level components or K8s related terms. Can we have an eventing subsection?

> ## Glossary for Eventing

Author: Added them as a new subsection, as discussed!
